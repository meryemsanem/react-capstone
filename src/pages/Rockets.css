.rockets-container {
  padding: 10px 5%;
}

.rocket-left {
  display: flex;
  gap: 20px;
  margin-bottom: 40px;
}

.rocket-image {
  width: 400px;
}

.rocket-btn {
  padding: 10px;
  background-color: #1c7ed6;
  color: white;
  border: none;
  border-radius: 4px;
  font-size: 20px;
  margin-top: 20px;
}

.rocket-name,
.rocket-desc {
  color: #333;
}
