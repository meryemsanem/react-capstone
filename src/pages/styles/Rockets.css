.rockets-container {
  padding: 10px 5%;
}

.rocket-left {
  display: flex;
  gap: 20px;
  margin-bottom: 40px;
}

.rocket-image {
  width: 400px;
}

.rocket-btn {
  padding: 10px;
  background-color: #1c7ed6;
  color: white;
  border: none;
  border-radius: 4px;
  font-size: 20px;
  margin-top: 20px;
}

.rocket-name,
.rocket-desc {
  color: #333;
}

.reserved-badge {
  padding: 1px 4px;
  border: 1px solid #22b8cf;
  background-color: #22b8cf;
  border-radius: 4px;
  font-size: 14px;
  margin-right: 10px;
  color: white;
}

.cancel {
  background-color: white;
  color: #777;
  border: 1px solid #777;
}
