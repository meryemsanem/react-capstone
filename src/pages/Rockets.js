import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  fetchRocketsData,
  reserveRocket,
  cancelReserveRocket,
} from '../redux/rockets/rocketsSlice';
import './Rockets.css';

const Rockets = () => {
  const dispatch = useDispatch();
  const { isLoading, data, error } = useSelector((state) => state.rockets);

  useEffect(() => {
    dispatch(fetchRocketsData());
  }, [dispatch]);

  const handleReserveButtonClick = (rocketId) => {
    dispatch(reserveRocket(rocketId));
  };

  const handleCancelReservationButtonClick = (rocketId) => {
    dispatch(cancelReserveRocket(rocketId));
  };

  if (isLoading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return <div>{error}</div>;
  }

  return (
    <div className="rockets-container">
      {data.map((rocket) => (
        <div className="rocket-left" key={rocket.id}>
          <img src={rocket.image} alt={rocket.name} className="rocket-image" />
          <div className="rocket-details">
            <h2 className="rocket-name">{rocket.name}</h2>
            <p className="rocket-desc">{rocket.description}</p>
            <button
              type="submit"
              className="rocket-btn"
              onClick={() => handleReserveButtonClick(rocket.id)}
            >
              Reserve Rocket
            </button>
            <button
              type="button"
              className="rocket-btn"
              onClick={() => handleCancelReservationButtonClick(rocket.id)}
            >
              Cancel Reservation
            </button>
          </div>
        </div>
      ))}
    </div>
  );
};

export default Rockets;
