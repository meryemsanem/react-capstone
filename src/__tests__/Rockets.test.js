import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Rockets from '../pages/Rockets';

const mockStore = configureStore([thunk]);

test('matches snapshot for Rockets component', () => {
  const store = mockStore({
    rockets: {
      isLoading: false,
      data: [
        {
          id: 1,
          name: 'Rocket 1',
          description: 'Description 1',
          image: 'rocket1.jpg',
        },
        {
          id: 2,
          name: 'Rocket 2',
          description: 'Description 2',
          image: 'rocket2.jpg',
        },
      ],
      joinedRockets: [1],
    },
  });

  const { asFragment } = render(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  );

  expect(asFragment()).toMatchSnapshot();
});
