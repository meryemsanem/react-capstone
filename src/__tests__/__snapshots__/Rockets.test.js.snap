// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for Rockets component 1`] = `
<DocumentFragment>
  <div
    class="rockets-container"
  >
    <div
      class="rocket-left"
    >
      <img
        alt="Rocket 1"
        class="rocket-image"
        src="rocket1.jpg"
      />
      <div
        class="rocket-details"
      >
        <h2
          class="rocket-name"
        >
          Rocket 1
        </h2>
        <p
          class="rocket-desc"
        >
           
          <span
            class="reserved-badge"
          >
            Reserved
          </span>
          Description 1
        </p>
        <button
          class="rocket-btn cancel"
          data-testid="cancel-reservation-button-1"
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </div>
    <div
      class="rocket-left"
    >
      <img
        alt="Rocket 2"
        class="rocket-image"
        src="rocket2.jpg"
      />
      <div
        class="rocket-details"
      >
        <h2
          class="rocket-name"
        >
          Rocket 2
        </h2>
        <p
          class="rocket-desc"
        >
          Description 2
        </p>
        <button
          class="rocket-btn"
          data-testid="reserve-button-2"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
